### PR TITLE
Make build release and copy to bin

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @shenli  @BusyJay @disksing @zhangjinpeng1987

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -6,10 +6,12 @@ GOROOT ?= $(DEPS_PATH)/go
 
 .PHONY: all
 
-all: format build test
+all: release
```
Comment (BusyJay) [inline Makefile]: I think reorder the target order is better.

Comment (siddontang) [inline Makefile]: we will use make to install the release into bin.

File: Makefile [hunk 2]
```diff
@@ -6,10 +6,12 @@ GOROOT ?= $(DEPS_PATH)/go
...
 
 dev:
-	@export ENABLE_FEATURES=dev && make
+	@export ENABLE_FEATURES=dev && make _dev
+
+_dev: format build test
```
Comment (BusyJay) [inline Makefile]: Why not just move the dependency targets to dev?

Comment (siddontang) [inline Makefile]: how to set `ENABLE_FEATURES=dev`? if we `make dev`?

File: Makefile (hunk 1 above)
Comment (BusyJay) [inline Makefile]: So just move the release target to the first one.

File: Makefile (hunk 2 above)
Comment (BusyJay) [inline Makefile]: If release target is the first rule, here we can just use `make all`.

File: Makefile (hunk 1 above)
Comment (siddontang) [inline Makefile]: do we need to set .PHONY?

Comment (disksing) [inline Makefile]: +1 for `make` => `make release`, `make all` => `make format+build+test`

**Comment by BusyJay** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**